src/solver/conv_ck_igemm_fwd_bias_activ.cpp

```diff
@@ -0,0 +1,445 @@
+    case miopenFloat:
+    case miopenInt32:
+    case miopenInt8x4:
+    case miopenBFloat16:
+    case miopenDouble: break;
+    }
+    return false;
+#endif
+}
+
+bool PerformanceConfigConvCKIgemmFwdBiasActiv::operator==(
+    const PerformanceConfigConvCKIgemmFwdBiasActiv& other) const
+{
+    return this->kernel_id == other.kernel_id;
+}
+PerformanceConfigConvCKIgemmFwdBiasActiv
+ConvCKIgemmFwdBiasActiv::GetDefaultPerformanceConfig(const FusionContext& ctx) const
+{
+    PerformanceConfigConvCKIgemmFwdBiasActiv pp;
+    pp.HeuristicInit(ctx);
+    return pp;
+}
+
+bool ConvCKIgemmFwdBiasActiv::IsValidPerformanceConfig(
+    const FusionContext& ctx, const PerformanceConfigConvCKIgemmFwdBiasActiv& config) const
+{
+    return config.IsValid(ctx);
+}
+
+PerformanceConfigConvCKIgemmFwdBiasActiv
+ConvCKIgemmFwdBiasActiv::Search(const FusionContext& ctx, const AnyInvokeParams& invoke_ctx) const
+{
+    return GenericSearch(*this, ctx, invoke_ctx);
+}
+
+bool ConvCKIgemmFwdBiasActiv::IsApplicable(const FusionContext& ctx) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    return false;
+#else
+    const auto& problem = ctx.problem.GetConvProblem(0, conv::Direction::Forward);
+    if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_CK_IGEMM_FWD_BIAS_ACTIV{}))
+        return false;
+    if(problem.conv_problem.GetConv().attribute.deterministic)
+        return false;
+    if(problem.conv_problem.GetInDataType() != problem.conv_problem.GetWeightsDataType() ||
+       problem.conv_problem.GetInDataType() != problem.conv_problem.GetOutDataType())
+        return false;
+    if(!problem.Is2d())
+        return false;
+    const std::string arch = ctx.GetStream().GetDeviceName();
+    if(arch != "gfx908" && arch != "gfx90a")
+        return false;
+    if(!problem.IsLayoutNHWC())
+        return false;
+
+    switch(problem.conv_problem.GetInDataType())
+    {
+    case miopenHalf: return CheckCKApplicability<ck::half_t>(problem);
+    case miopenInt8:
+    case miopenFloat:
+    case miopenInt32:
+    case miopenInt8x4:
+    case miopenBFloat16:
+    case miopenDouble: break;
+    }
+    return false;
+#endif
+}
+
+ConvSolution
+ConvCKIgemmFwdBiasActiv::GetSolution(const FusionContext& ctx,
+                                     const PerformanceConfigConvCKIgemmFwdBiasActiv& config) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    std::ignore = config;
+    return {};
+#else
+    const auto& problem = ctx.problem.GetConvProblem(0, conv::Direction::Forward);
+    ConvSolution result;
+    result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
+        std::ignore = kernels;
+        return [=](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+            switch(problem.conv_problem.GetInDataType())
+            {
+            case miopenHalf:
+                RunCKSolution<ck::half_t>(handle, primitive_parameters, problem, config);
```

Comment:
Are there no instances for FP32 at all ?

Reply:
Currently only CK is only supporting FP16 https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/develop/library/src/tensor_operation_instance/gpu/conv2d_fwd_bias_relu/device_conv2d_fwd_xdl_c_shuffle_bias_relu_nhwc_kyxc_nhwk_f16_instance.cpp